File: 2-hashing/largestUniqueNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 * 
 * Time: O(N)
 * Space: O(N)
 */
var largestUniqueNumber = function(nums) {
  const freq = new Map();
  for (const num of nums) {
      freq.set(num, (freq.get(num) || 0) + 1);
  }
  console.log(freq)
  let max = -1;
  for (const [num, count] of freq) {
      if (count === 1) max = Math.max(num, max);
  }
  return max;
};

File: 2-hashing/largestUniqueNumber.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        # Time: O(N)
        # Space: O(N)
        freq = {};
        for num in nums:
            freq[num] = freq.get(num, 0) + 1
        ans = -1
        for num, count in freq.items():
            if count == 1:
                ans = max(num, ans)
        return ans
    